docs/manual_api/edsl.rst

```diff
@@ -34,22 +34,61 @@ Objects
    .. group-tab:: Python
 
       .. autoclass:: plaidml.edsl.IndexedTensor
+         :members:
+         :special-members:
       .. autoclass:: plaidml.edsl.LogicalShape
          :members:
       .. autoclass:: plaidml.edsl.Tensor
+         :members:
       .. autoclass:: plaidml.edsl.TensorRef
+         :members:
       .. autoclass:: plaidml.edsl.ProgramArgument
+         :members:
       .. autoclass:: plaidml.edsl.Program
+         :members:
       .. autoclass:: plaidml.edsl.TensorDim
+         :members:
       .. autoclass:: plaidml.edsl.TensorIndex
+         :members:
       .. autoclass:: plaidml.edsl.Constraint
+         :members:
 
 ----------
 Primitives
 ----------
 
-.. doxygengroup:: edsl_primitives
-   :content-only:
+.. tabs::
+   .. group-tab:: C++
+
+      .. doxygengroup:: edsl_primitives
+         :content-only:
+
+   .. group-tab:: Python
+
+      .. autofunction:: plaidml.edsl.abs
+      .. autofunction:: plaidml.edsl.as_float
+      .. autofunction:: plaidml.edsl.as_int
+      .. autofunction:: plaidml.edsl.as_uint
+      .. autofunction:: plaidml.edsl.as_bool
```

Comment:
I'd like to deprecate these older `as_XXX` casting methods with a `cast` method. For instance, we have no good way to cast between a `BF16` and an `FP32` using `as_float()`